File: backend/core-services/product-service/src/main/java/com/movieticket/product/entity/ShowSchedule.java
package com.movieticket.product.entity;

import com.movieticket.product.enums.ProjectionType;
import com.movieticket.product.enums.TranslationType;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "show_schedules")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ShowSchedule {
  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private String id;

  @ManyToOne
  @JoinColumn(name = "movie_id")
  private Movie movie;

  @Enumerated(EnumType.STRING)
  private ProjectionType projectionType;

  @Enumerated(EnumType.STRING)
  private TranslationType translationType;

  private String roomId;
  private LocalDateTime startTime;
  private LocalDateTime endTime;
  private int availableSeat;
  private boolean status = true;

  private LocalDateTime createdAt;
  private LocalDateTime updatedAt;

  private String cinemaId;

  @PrePersist
  protected void onCreate() {
    createdAt = LocalDateTime.now();
    updatedAt = LocalDateTime.now();
  }

  @PreUpdate
  protected void onUpdate() {
    updatedAt = LocalDateTime.now();
  }
}

File: frontend/web/src/types/show-schedule.ts
export interface ShowScheduleResDTO {
    id: string;
    movieId: string;
    movieTitle: string;
    movieThumbnailUrl: string;
    projectionType: string;
    translationType: string;
    roomId: string;
    cinemaId: string;
    startTime: string;
    endTime: string;
    availableSeat: number;
    status: boolean;
}

export interface ShowScheduleSearchDTO {
    movieId?: string;
    cinemaId?: string;
    roomId?: string;
    projectionType?: string;
    translationType?: string;
    status?: boolean;
    date?: string;
    page?: number;
    size?: number;
}

export interface Showtime {
    id: string;
    time: string;
}

export interface FormatShowtime {
    projection: string;
    translation: string;
    showtimes: Showtime[];
}

export interface CinemaShowtime {
    cinemaId: string;
    cinemaName: string;
    formats: FormatShowtime[];
}